src/AddIns/Uno.WinUI.Graphics/GLCanvasElement.WinUI.cs

```diff
@@ -0,0 +1,280 @@
+				}
+				throw new InvalidOperationException("ChoosePixelFormat failed");
+			}
+
+			_glContext = NativeMethods.wglCreateContext(_hdc);
+
+			if (_glContext == IntPtr.Zero)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error($"wglCreateContext failed");
+				}
+				throw new InvalidOperationException("ChoosePixelFormat failed");
+			}
+		}
+
+		public object CreateGLSilkNETHandle() => GL.GetApi(new WpfGlNativeContext());
+
+		public void DestroyContext()
+		{
+			NativeMethods.wglDeleteContext(_glContext);
+			_glContext = default;
+			_hdc = default;
+		}
+
+		public IDisposable MakeCurrent()
+		{
+			var glContext = NativeMethods.wglGetCurrentContext();
+			var dc = NativeMethods.wglGetCurrentDC();
+			NativeMethods.wglMakeCurrent(_hdc, _glContext);
+			return Disposable.Create(() => NativeMethods.wglMakeCurrent(dc, glContext));
+		}
+
+		// https://sharovarskyi.com/blog/posts/csharp-win32-opengl-silknet/
+		private class WpfGlNativeContext : INativeContext
+		{
+			private readonly UnmanagedLibrary _l;
+
+			public WpfGlNativeContext()
+			{
+				_l = new UnmanagedLibrary("opengl32.dll");
+				if (_l.Handle == IntPtr.Zero)
+				{
+					throw new PlatformNotSupportedException("Unable to load opengl32.dll. Make sure you're running on a system with OpenGL support");
+				}
+			}
+
+			public bool TryGetProcAddress(string proc, out nint addr, int? slot = null)
+			{
+				if (_l.TryLoadFunction(proc, out addr))
+				{
+					return true;
+				}
+
+				addr = NativeMethods.wglGetProcAddress(proc);
+				return addr != IntPtr.Zero;
+			}
+
+			public nint GetProcAddress(string proc, int? slot = null)
+			{
+				if (TryGetProcAddress(proc, out var address, slot))
+				{
+					return address;
+				}
+
+				throw new InvalidOperationException("No function was found with the name " + proc + ".");
+			}
+
+			public void Dispose() => _l.Dispose();
+		}
+	}
+
+	// https://sharovarskyi.com/blog/posts/csharp-win32-opengl-silknet/
+	private class WinUINativeContext : INativeContext
+	{
+		private readonly UnmanagedLibrary _l;
+
+		public WinUINativeContext()
+		{
+			_l = new UnmanagedLibrary("opengl32.dll");
+			if (_l.Handle == IntPtr.Zero)
+			{
+				throw new PlatformNotSupportedException("Unable to load opengl32.dll. Make sure you're running on a system with OpenGL support");
+			}
+		}
+
+		public bool TryGetProcAddress(string proc, out nint addr, int? slot = null)
+		{
+			if (_l.TryLoadFunction(proc, out addr))
+			{
+				return true;
+			}
+
+			addr = NativeMethods.wglGetProcAddress(proc);
+			return addr != IntPtr.Zero;
+		}
+
+		public nint GetProcAddress(string proc, int? slot = null)
+		{
+			if (TryGetProcAddress(proc, out var address, slot))
+			{
+				return address;
+			}
+
+			throw new InvalidOperationException("No function was found with the name " + proc + ".");
+		}
+
+		public void Dispose() => _l.Dispose();
+	}
+
+	private static class NativeMethods
+    {
+    	[DllImport("user32.dll")]
```

Comment:
Let's switch to LibraryImport

Reply:
The LibraryImport for these declarations is just a nested DllImport, so it's unnecessary here.